catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SortedQueryMonitor.java

```diff
@@ -127,18 +136,18 @@ public void run() {
                     }
                 }
 
-                source = futures.remove(future);
-                if (source != null) {
-                    sourceId = source.getId();
-                }
+                queryRequest = futures.remove(future);
+                sourceId = getSourceIdFromRequest(queryRequest);
 
-                SourceResponse sourceResponse = future.get();
+                sourceResponse = future.get();
 
                 if (sourceResponse == null) {
                     LOGGER.debug("Source {} returned null response", sourceId);
                     processingDetails.add(new ProcessingDetailsImpl(sourceId,
```

Comment:
I think we should add this `ProcessingDetails` to a `SourceResponse` and send it to `executePostFederationQueryPlugins`.  Sense we are doing this in all error cases, I think you can extract some common code that creates the processing details and sends it to `executePostFederationQueryPlugins` and adds it to `processingDetails`.